code/modules/economy/quikpay.dm

```diff
@@ -97,12 +97,30 @@
 
 /obj/item/device/nanoquikpay/attackby(obj/O, mob/user)
 	var/obj/item/card/id/I = O.GetID()
+	
+	if (istype(O, /obj/item/spacecash/ewallet))
```

Comment:
put this bit above the part where you check if it's an ID, and add a return after the `please return to home screen` bit